File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/CheckedRule.java
package com.wesleyelliott.kubwa.rule;

/**
 * Created by wesley on 2016/07/28.
 */

public class CheckedRule extends Rule<Boolean> {

    private boolean value;

    public CheckedRule(Boolean value) {
        super(Boolean.class);
        this.value = value;
    }

    @Override
    public boolean isValid(Boolean value) {
        return value != null && this.value == value;
    }
}

File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/CheckedRule.kt
package com.wesleyelliott.kubwa.rule

/**
 * Created by wesley on 2016/07/28.
 */

class CheckedRule(private val value: Boolean?) : Rule<Boolean>(Boolean::class.java) {

    override fun isValid(value: Boolean?): Boolean {
        return value != null && this.value == value
    }
}
